packages/@aws-cdk/aws-glue-alpha/lib/jobs/scala-spark-etl-job.ts

```diff
@@ -8,5 +8,176 @@
  *  —enable-metrics, —enable-spark-ui, —enable-continuous-cloudwatch-log.
  *  You can find more details about version, worker type and other features
  *  in Glue's public documentation.
+ *
+ *  RFC: https://github.com/aws/aws-cdk-rfcs/blob/main/text/0497-glue-l2-construct.md
+ *
  */
 
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { Bucket } from 'aws-cdk-lib/aws-s3';
+import { CfnJob } from 'aws-cdk-lib/aws-glue';
+import { Job, JobProperties } from './job';
+import { Construct } from 'constructs';
+import { JobType, GlueVersion, JobLanguage, WorkerType } from '../constants';
+import { SparkUIProps, SparkUILoggingLocation, validateSparkUiPrefix, cleanSparkUiPrefixForGrant } from './spark-ui-utils';
+import { Code } from '../code';
+
+/**
+ * Properties for creating a Scala Spark ETL job
+ */
+export interface ScalaSparkEtlJobProps extends JobProperties {
```

Comment:
What is the benefit to create a separate class for PySpark and Scala?

Reply:
The goal for the re-architecture of the Glue L2 construct is to enforce defaults and parameters via contracts and interfaces rather than the sync-time validations that currently exist in the construct. Having separate classes allow us to inherit the common aspects of the jobs, but enforce the particular differences via the job-specific interfaces.

Reply:
The end goal is to allow for a better developer experience making it absolutely clear what's required and what's not needed / won't be used at development time rather than at synth time.